File: frontend/src/types/source.ts
export interface Source {
  id: string;
  name: string;
  slug: string;
  base_url: string;
  rss_feeds: string[];
  scraper_type: string;
  category: string;
  is_active: boolean;
  scrape_interval_minutes: number;
  rate_limit_rpm: number;
  has_paywall: boolean;
  last_scraped_at: string | null;
  created_at: string;
  updated_at: string;
  articles_today: number;
  total_articles: number;
  last_run_status: string | null;
}

export interface ScrapeRun {
  id: string;
  source_id: string;
  status: string;
  articles_found: number;
  articles_new: number;
  articles_updated: number;
  error_message: string | null;
  started_at: string;
  completed_at: string | null;
  duration_seconds: number | null;
  created_at: string;
  source_name: string | null;
  source_slug: string | null;
}

export interface DashboardStats {
  total_articles: number;
  articles_today: number;
  active_sources: number;
  total_sources: number;
  last_scrape_at: string | null;
  articles_by_source: { name: string; slug: string; count: number }[];
  articles_by_category: { category: string; count: number }[];
  recent_runs: ScrapeRun[];
}

File: backend/app/schemas/source.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class RssFeedEntry(BaseModel):
    """Kategorize edilmis RSS feed girisi: {"url": "...", "category": "..."}"""
    url: str
    category: str | None = None


class SourceBase(BaseModel):
    name: str
    slug: str
    base_url: str
    # Geriye donuk uyumluluk: eski format str listesi, yeni format dict listesi
    rss_feeds: list[RssFeedEntry | str] = []
    scraper_type: str = "rss"
    category: str = "general"
    is_active: bool = True
    scrape_interval_minutes: int = 60
    rate_limit_rpm: int = 10
    has_paywall: bool = False
    config: dict | None = None


class SourceCreate(SourceBase):
    pass


class SourceUpdate(BaseModel):
    is_active: bool | None = None
    scrape_interval_minutes: int | None = None
    rate_limit_rpm: int | None = None
    rss_feeds: list[RssFeedEntry | str] | None = None
    scraper_type: str | None = None
    config: dict | None = None


class SourceResponse(SourceBase):
    id: uuid.UUID
    last_scraped_at: datetime | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class SourceWithStats(SourceResponse):
    articles_today: int = 0
    total_articles: int = 0
    last_run_status: str | None = None
